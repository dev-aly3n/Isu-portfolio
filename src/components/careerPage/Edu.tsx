// libs
import { useState } from "react";
import { motion } from "framer-motion";
// components
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faAngleRight } from "@fortawesome/free-solid-svg-icons";
import { useOnScrollAnimation } from "../../hooks/useOnScrollAnimation";
import { secFading } from "../../utils/animation";
// hooks
// store
// utils & animation

interface props {
  education: { name: string; field: string; about: string; image: string };
}

const Edu: React.FC<props> = ({ education }) => {
  const [isExpanded, setIsExpanded] = useState(false);
  const [sec,controls] = useOnScrollAnimation();

  return (
    <motion.div 
    ref={sec}
    variants={secFading}
    initial="hidden"
    animate={controls}
    
    className="flex flex-col my-3 w-11/12 sm:w-full ">
      <div
        className="no-select-a-tag flex flex-col sm:flex-row justify-start items-center hover:cursor-pointer hover:bg-gray-500/5 duration-1000 relative"
        onClick={() => setIsExpanded((prevState) => !prevState)}
      >
        <motion.span layout className="text-3xl absolute left-0 top-[50%] mt-[-18px] flex justify-center items-center">
          <FontAwesomeIcon
            className={`duration-500 ${isExpanded ? "rotate-90" : "rotate-0"}`}
            icon={faAngleRight}
          />{" "}
        </motion.span>
        <motion.img
          layout
          src={`/media/logos/${education.image}.png`}
          className="block w-[100px] md:w-[10vw] ml-4"
        />
        <motion.div layout className="flex flex-col ml-5 sm:ml-2">
          <h3 className="text-2xl font-semibold mb-10">{education.name}</h3>
          <p>{education.field}</p>
        </motion.div>
      </div>
      <motion.div layout className="w-full h-0.5 bg-white" />
      <motion.div
        layout
        className={`flex flex-col mx-2 my-3 px-5 overflow-hidden ${
          isExpanded ? "h-full" : "h-0"
        }`}
      >
        {education.about}
      </motion.div>
    </motion.div>
  );
};

export default Edu;
