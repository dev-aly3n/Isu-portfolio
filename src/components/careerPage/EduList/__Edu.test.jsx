import { render, screen } from "@testing-library/react";
import Edu from "./Edu";
import { skillMock } from "../../../__mock__";

const education = {
  name: "uni of teh",
  field: "mechanics",
  about: "something about uni",
  image: "/media/logos/kntu.png",
};

describe("career/Skill", () => {
  it("skill snap", () => {
    const { container } = render(<Edu education={education} />);
    expect(container).toMatchSnapshot();
  });
});
