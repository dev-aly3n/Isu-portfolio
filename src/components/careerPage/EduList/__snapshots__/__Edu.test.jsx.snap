// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`career/Skill skill snap 1`] = `
<div>
  <div
    class="edu-container"
    style="opacity: 0;"
  >
    <div>
      <span>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-right fa-w-8 duration-500 rotate-0"
          data-icon="angle-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 256 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            fill="currentColor"
          />
        </svg>
         
      </span>
      <span>
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
          />
          <noscript />
        </span>
      </span>
      <div>
        <h3>
          uni of teh
        </h3>
        <p>
          mechanics
        </p>
      </div>
    </div>
    <div />
    <div
      class=" h-0"
    >
      something about uni
    </div>
  </div>
</div>
`;
